### ISSUE TITLE
Out of Memory (NVIDIA GeForce RTX 3070 Ti) - MDX23C

### ISSUE THREAD

**Issue by KatieBelli** [ZFTurbo/Music-Source-Separation-Training#123]
Is it even possible to train an MDX23C model with an NVIDIA GeForce RTX 3070 Ti? 
I always get this error:

![Image](https://github.com/user-attachments/assets/0ab370d5-5edd-4cc6-85c0-8b7a0a592d0b)

I used batch size 1 and gradient_accumulation_steps 6 _(but the loss and SDR didn't change after training epoch 0 and epoch 1)_ but the losses didn't change so I adjusted the batch size to 2 and gradient_accumulation_steps to 3 but this change causes OOM.

_My current config file where it tells me OOM:_

```
audio:
  chunk_size: 261120
  dim_f: 4096
  dim_t: 256
  hop_length: 1024
  n_fft: 8192
  num_channels: 2
  sample_rate: 44100
  min_mean_abs: 0.001

model:
  act: gelu
  bottleneck_factor: 4
  growth: 128
  norm: InstanceNorm
  num_blocks_per_scale: 2
  num_channels: 128
  num_scales: 5
  num_subbands: 4
  scale:
  - 2
  - 2

training:
  batch_size: 2
  gradient_accumulation_steps: 3
  grad_clip: 0
  instruments:
  - .............................................
  - .............................................
  lr: 9.0e-05
  patience: 2
  reduce_factor: 0.95
  target_instrument: null
  num_epochs: 1000
  num_steps: 1000
  q: 0.95
  coarse_loss_clip: true
  ema_momentum: 0.999
  optimizer: adam
  other_fix: false # it's needed for checking on multisong dataset if other is actually instrumental
  use_amp: true # enable or disable usage of mixed precision (float16) - usually it must be true****
```

I already added this to the training script but it doesn't help:

```
os.environ['PYTORCH_CUDA_ALLOC_CONF'] = 'expandable_segments:True' 

def clear_gpu_memory(): torch.cuda.empty_cache() def clear_gpu_memory(): torch.cuda.empty_cache() 
```

_My data structure (flac files, hope this structure is alright):_

```
training
for each song one folder
in it: 2 separate files but no mixture of them
```

```
validation
for each song one folder
in it: 2 separate files (which are synced) + them mixed together in one file
```

Overall length of all training files: ~24h (226 songs)
Overall length of all validation files: ~9h (57 songs)
(Maybe I should use less training/validation files?)

Should I add [this](https://stackoverflow.com/questions/73747731/runtimeerror-cuda-out-of-memory-how-can-i-set-max-split-size-mb) to my training script?

**If my GPU isn't enough, please recommend a cloud GPU which is completely free, stronger than NVIDIA GeForce RTX 3070 Ti and where I don't need to pay for my dataset to upload to a cloud storage (because it's over 8GB and would take alot of time to upload + some might not even allow to upload over 8GB of data).**

PS: I use the GUI MSST-GUI.exe with num workers 8.

**Comment by KatieBelli** [comment]
Did some things and now it works:

- I increased the pagefile size manually in the system settings
- removed some songs from my training dataset (but maybe that wasn't the cause)
